src/meta_tags.js

```diff
@@ -56,6 +56,9 @@ class MetaTags extends Component {
         } else if (tag === 'meta') {
           const meta = getDuplicateMeta(child);
           if (meta) removeChild(head, meta);
+        } else if (tag === 'link' && child.rel === 'canonical') {
+          const link = getDuplicateMeta(child);
```

Comment:
getDuplicateMeta will return null if no id is defined on it. As it id, name and property attribute

Reply:
I fixed  getDuplicateMeta in my fork, please check below commit
https://github.com/usefksa/react-meta-tags/commit/7f2fb4210f7cc6faa09ffb7b78f0f1d656c69651